common/static/common/js/utils/edx.utils.validate.js

```diff
@@ -102,6 +136,14 @@
                             return isBlank;
                         },
 
+                        // To support login by username
+                        username: {
+                            // In general case, username can only contain letters (A-Z, a-z), numerals (0-9), underscores (_), and hyphens (-)
+                            ascii_regex: new RegExp('^([A-Za-z0-9-_]+)$'),
+                            // If the `ENABLE_UNICODE_USERNAME` feature enabled, username can only contain letters, numerals, and @/./+/-/_ characters
+                            // unicode_regex: new RegExp('^([A-Za-z0-9_@.+-]+)$'),
```

Comment:
Why unicode_regex is commented?

Reply:
Demoed the case with the unicode to Alex, but the ENABLE_UNICODE_USERNAME feature is disabled. I've actually never seen it enabled, but it's there.